src/main/scala/uk/gov/hmrc/sbtdistributables/SbtDistributablesPlugin.scala

```diff
@@ -114,18 +116,19 @@ object SbtDistributablesPlugin extends AutoPlugin {
     outputStream.closeArchiveEntry()
   }
 
-  private def copyEntries(inputStream: ZipInputStream, outputStream: TarArchiveOutputStream, root: File, artifactName: String) {
+  private def copyEntries(inputStream: ZipInputStream, outputStream: TarArchiveOutputStream, root: File, artifactName: String, executableFiles: Seq[File]) {
     var inputZipEntry: ZipEntry = inputStream.getNextEntry
     while (inputZipEntry != null) {
-      outputStream.putArchiveEntry(tarArchiveEntry(root, inputZipEntry.getName, inputZipEntry.getSize, inputZipEntry.getTime, getTarEntryMode(inputZipEntry.getName, artifactName)))
+      outputStream.putArchiveEntry(tarArchiveEntry(root, inputZipEntry.getName, inputZipEntry.getSize, inputZipEntry.getTime, getTarEntryMode(inputZipEntry.getName, artifactName, executableFiles)))
       copy(inputStream, outputStream)
       outputStream.closeArchiveEntry()
       inputZipEntry = inputStream.getNextEntry
     }
   }
 
-  private def getTarEntryMode(zipEntryName: String, artifactName: String): Option[Int] = {
-    if (zipEntryName.endsWith(s"/bin/$artifactName")) {
+
+  private def getTarEntryMode(zipEntryName: String, artifactName: String, executableFiles: Seq[File]): Option[Int] = {
+    if (zipEntryName.endsWith(s"/bin/$artifactName") || executableFiles.map(_.name).contains(zipEntryName)) {
```

Comment:
We ask for a sequence of `java.io.File` but only use names, so a string would do here.